frontend/packages/pipelines-plugin/src/components/catalog/apis/__tests__/tektonHub.spec.ts

```diff
@@ -0,0 +1,95 @@
+import { coFetch } from '@console/internal/co-fetch';
+import { sampleTektonHubCatalogItem } from '../../../../test-data/catalog-item-data';
+import {
+  TektonHubTaskVersion,
+  getApiResponse,
+  getHubUIPath,
+  TEKTON_HUB_ENDPOINT,
+} from '../tektonHub';
+
+const coFetchMock = coFetch as jest.Mock;
+
+jest.mock('@console/internal/co-fetch', () => ({
+  coFetch: jest.fn(),
+}));
+
+const apiResults = ({
+  ok,
+  status,
+  versions,
+}: {
+  ok: boolean;
+  status?: number;
+  versions?: TektonHubTaskVersion[];
+}) =>
+  Promise.resolve({
+    ok,
+    ...(status && { status }),
+    json: () => ({
+      data: {
+        versions,
+      },
+    }),
+  });
+
+beforeEach(() => {
+  coFetchMock.mockClear();
+  coFetchMock.mockReturnValue(
+    apiResults({ ok: true, status: 200, versions: sampleTektonHubCatalogItem.attributes.versions }),
+  );
+});
+
+afterEach(() => {
+  coFetchMock.mockReset();
+});
+
+describe('getApiResponse', () => {
+  it('should throw error if the response status is not 200', async () => {
+    coFetchMock.mockReturnValue(
+      apiResults({
+        ok: true,
+        status: 404,
+        versions: sampleTektonHubCatalogItem.attributes.versions,
+      }),
+    );
+    try {
+      await getApiResponse('testurl');
+    } catch (e) {
+      expect(e.message).toBe('Api response failed with status: 404');
+    }
```

Comment:
You removed the done here, which is great because the done is usually used with callbacks.

But when you expect an error it is a good idea to fail also if there was no error, otherwise you might not notice that this method doesn't return 404 anymore.

```suggestion
      fail("Expect an error if the response status is not 200")
    } catch (e) {
      expect(e.message).toBe('Api response failed with status: 404');
    }
```

Reply:
You are right, the test case will not catch the errors if the method implementation is changed for some reason. Added it 👍